recipe/build.sh

```diff
@@ -37,7 +37,7 @@ elif [[ "$target_platform" == osx-* ]]; then
   # This is patched by the linker packaged in conda, but when the environment is
   # not activated, clang tries to use the system linker.
   # Adding -B $PREFIX/libexec/pocl makes clang look there for the linker first.
-  EXTRA_HOST_LD_FLAGS="$EXTRA_HOST_LD_FLAGS -undefined dynamic_lookup -B $PREFIX/libexec/pocl"
+  EXTRA_HOST_LD_FLAGS="$EXTRA_HOST_LD_FLAGS -undefined dynamic_lookup -B $PREFIX/libexec/pocl -L$SDKROOT/usr/lib -lSystem"
```

Comment:
SDKROOT is different in different machines. This would bake the build machine's PATH into the pocl binary.

Reply:
I see. Is there anything else we can do? Is this an issue in the recent ld64=711 release?

Reply:
We might need to add a function like the following to get the sdkroot at https://github.com/conda-forge/cctools-and-ld64-feedstock/blob/522ce9e43aae12b7601e40034fc5d757cb21fe10/recipe/patches/0002-ld64-add-conda-specific-env-vars-to-modify-lib-searc.patch#L41
```
#include <cstdio>
#include <iostream>
#include <memory>
#include <stdexcept>
#include <string>
#include <array>

std::string sdkroot() {
    std::string cmd = "xcrun --show-sdk-path";
    std::array<char, 128> buffer;
    std::string result;
    std::unique_ptr<FILE, decltype(&pclose)> pipe(popen(cmd.c_str(), "r"), pclose);
    if (!pipe) {
        throw std::runtime_error("popen() failed!");
    }
    while (fgets(buffer.data(), static_cast<int>(buffer.size()), pipe.get()) != nullptr) {
        result += buffer.data();
    }
    if (!result.empty() && result[result.length()-1] == '\n') {
        result.erase(result.length()-1);
    }
    return result;
}
```